core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java

```diff
@@ -1431,4 +1433,32 @@ private boolean rerunWF(String workflowId, String taskId, Map<String, Object> ta
         }
         return false;
     }
+
+    public void removeLoopOverTasks(Task loopTask, Workflow workflow) {
+        // Remove all tasks after the "DO_WHILE" task
+        for(Task task: workflow.getTasks()) {
```

Comment:
@apanicker-nflx, @kishorebanala, @mudit3774, @naveenchlsn Even after removing task after every iteration I still see that workflow has the task already. This happens because any of loopover task will get updated in executionDAO after do while task gets updated. I am scheduling task immediately after removing from executionDAO and so I am getting all the task from task mapper properly. I feel this is not the good way of doing it. We have to remove from executionDAo otherwise scheduling second iteration will be a problem. Suggestion are welcomed. On approach I see is to rename a task with iteration and store it. But in that case also original task must be removed.

Reply:
Have you considered an approach similar to the `FORK..JOIN` concept which constitutes two system tasks?
Essentially, have `DO...WHILE` as two separate system tasks. 
1. DO - This takes a list of tasks that need to be executed in a loop as the input and is executed only once.
2. WHILE - This takes an expression to be evaluated as input. It could also have a configurable flag to control if iteration history needs to be saved. This task could be executed multiple times.

When the workflow is run, the `DO` task moves directly to `COMPLETED` state and schedules a `WHILE` task. The `WHILE` task could then schedule the first task in the list or complete execution based on the condition evaluation. It can also decide on saving iteration history based on the flag. This approach could probably alleviate some of the hurdles. 
Let me know your thoughts.

Reply:
Hi @apanicker-nflx . Thanks for the suggestion. 
1) If we go with DO...WHILE , still the issue to schedule second iteration will be problem unless I change the task reference name in every iteration and schedule it. Issue is, in second iteration even though task mapper gives correct task but it will be omitted in DeciderService.getTasksToBeScheduled because workflow will have that task with same reference name.
2) In case of changing the task reference name and scheduling it we have to maintain the list of task that we have scheduled last and wait for all of them to get completed before scheduling second set. Consider a case like DO..WHILE (HTTP1,HTTP2). In this case while scheduling second iteration, we will schedule HTTP1_2 forcefully (Create new task object). When it gets completed its next task will be null (ideally we can make HTTP2 but then we have to construct task object). So when execute of WHILE will get called we have to know that HTTP1 is completed and now time to schedule HTTP2. So we have to store list of last scheduled tasks.
To solve this problem we have to change the way DeciderService.getTasksToBeScheduled works.

Reply:
Hi @apanicker-nflx Let me know your thoughts on this. Let's close this as soon as possible.

Reply:
Hi @manan164 Thanks for the response. 
1. I feel that since we are violating the DAG structure for a workflow with the concept of loops, we should introduce a new field `iteration` on the task which can be used alongwith `taskReferenceName` to ensure that duplicate tasks are not scheduled within an execution. The logic in the task mappers and `tasksToBeScheduled` would then need to be updated to account for this. 
2. This new field also enables the re-use of the current logic for finding the next task from a given task by not changing the `taskReferenceName`, but rather updating the iteration. 

Let me know if this approach sounds reasonable.

Reply:
Hi @apanicker-nflx Iteration is already there https://github.com/Netflix/conductor/pull/1169/files#diff-5ef655e1d0e0f90d2cbc1edca630bae4R193. But currently exposed only for dowhile task. 
There are two ways to solve this.
1) While scheduling second iteration the task ref names will be like ``` taskRefName_iteration```. So while checking condition before scheduling next iteration, we will strip ```_iteration``` in case loopOver task is used in loopCondition. Also while getting the next task I will strip down ```iteration```. In this case I have to book keep what are the task scheduled currently as a part of second iteration. This is what I was mentioning in my previous comment.
2) Do not change task ref name in second iteration. Instead change the way taskMapper gives the task to DeciderService.getTasksToBeScheduled and change the executionDAO.createTasks to check iteration also. Currently it checks retry count only and that is the reason we update retryCount every time task is rescheduled. In this case also I need the book keeping mentioned in point 1.
Will create working model for point 1 so that things will be clear.

Reply:
Hi @apanicker-nflx I have made changes with the approach mentioned in point 1. Used delimiter ```:__:```. So task will be scheduled as taskRefName:__:iteration. Same will be shown in task details.  I have avoided book keeping list with the help of changing the DeciderService.getNextTask. Please have a look. Some nit picks are there but will change once we finalise the approach.

Reply:
@manan164 This approach looks good and very clean. Please fix the failing tests and add sufficient unit tests to cover all edge cases. Thanks a lot for your contribution.